### PR TITLE
don't transition leader/follower until previous one is done

### LINKED ISSUE #2355: RaceCondition on transition to leader directly after becoming follower
Each node becomes follower as default. It can happen that the leader election is fast and the transaction to leader is directly called afterwards, which means the follower partition is not yet installed and can't be removed since `startContext.hasService(followerPartitionServiceName)` will not return true in `removeFollowerPartitionService`.

Installation of a partition is async:
```java

  private void installFollowerPartition() {
    LOG.debug(
        "Installing follower partition service for partition {}", partitionInfo.getPartitionId());
    final Partition partition =
        new Partition(atomix.getEventService(), partitionInfo, RaftState.FOLLOWER);

    startContext
        .createService(followerPartitionServiceName, partition)
        .dependency(logStreamServiceName, partition.getLogStreamInjector())
        .dependency(stateStorageFactoryServiceName, partition.getStateStorageFactoryInjector())
        .group(FOLLOWER_PARTITION_GROUP_NAME)
        .install();
  }
```

This means the leader partition and follower partition is installed. Found this on implementing #2344 , in this case we have a leader which gets also the replicated snapshots, which is not intended.

### PR DESCRIPTION
**Pull request by deepthidevaki**
closes #2355

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/BrokerLeaderChangeTest.java [hunk 1]
```diff
@@ -135,6 +135,50 @@ public void shouldReactToLeaderChanges() {
...
+    final LeaderElection<String> election = getElection(getBroker(0), partition).getElection();
+
+    // when
+
+    // Force repeated leader change
+    election.async().anoint("0");
+    election.async().anoint("1");
+    election.async().anoint("2");
+
+    assertThat(election.getLeadership().leader().id()).isEqualTo(String.valueOf(2));
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/BrokerLeaderChangeTest.java]: The assertion is a bit flaky and also the assertions below

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Thanks for fixing this :+1: some small things

**Comment by deepthidevaki** [comment]
@zelldon I have updated PR as we discussed. Topology is updated only after leader/follower services are installed. In addition, before transitionFuture in PartitionInstallService was completed when the install is complete, now it also waits for the removals. Also moved all the services for leader into a composite service.

**Review by ChrisKujawa — APPROVED** [review]

**Comment by deepthidevaki** [comment]
bors retry

**Pull request merged**